pandas/core/indexes/range.py

```diff
@@ -64,6 +66,8 @@ class RangeIndex(Int64Index):
     _typ = 'rangeindex'
     _engine_type = libindex.Int64Engine
 
+    # check whether self._data has benn called
```

Comment:
do you actually need this?

Reply:
For some performance sensitive uses, it can be important to know whether the int64 array is being created or not and this attribue allows advanced users to check for it. A further use, which would perhaps make the actual structure of RangeIndex be more transparant for users, would be to use to conditionally add the byte size of the int64 array to ``.nbytes``

I will though try to remove the uses of ._data in many cases. E.g. it will be possible to have RangeIndex-specific  versions of ``__eq__``, ``__gt__`` etc. so probably the only need for calling ``_data`` would be when an boolean array is supplied to ``__getitem__``.

Reply:
Thoughts?

Reply:
do you actually use it though?

Reply:
I'm using it to check if various operations trigger calls to ._data. I don't use is outside of getting rid of the calls to _data.  

I'm thinking about eventually change ``nbytes`` to 

```python
    def nbytes(self):
        num_bytes = sum(getsizeof(getattr(self, v), 24) for v in
                   ['_start', '_stop', '_step'])
        if self._has_called_data:
            num_bytes += self._data.nbytes
        return num_bytes
```
+ add some docs on this behaviour.

But I also want to make Pandas call ._data in a lot fewer case, before I add the above.

Reply:
as I said, simply monkey patch what you need in the tests; adding this to the class is not a great idea 

OR maybe easier is to do the caching internally, e.g.

```
@property
def _data():
     if self._cached_data is None:
         self._cached_data = ...
     return self._cached_data
```

and set ``_cached_data = None`` in the constructor (this is pretty much what `@cache_readonly` does

Reply:
Ok, a ``_cached_data`` attribute would be fine by me and easy to test againts.